internal/appliance/reconciler/codeintel.go

```diff
@@ -0,0 +1,247 @@
+package reconciler
+
+import (
+	"context"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/sourcegraph/sourcegraph/internal/appliance/config"
+	"github.com/sourcegraph/sourcegraph/internal/k8s/resource/configmap"
+	"github.com/sourcegraph/sourcegraph/internal/k8s/resource/container"
+	"github.com/sourcegraph/sourcegraph/internal/k8s/resource/pod"
+	"github.com/sourcegraph/sourcegraph/internal/k8s/resource/pvc"
+	"github.com/sourcegraph/sourcegraph/internal/k8s/resource/secret"
+	"github.com/sourcegraph/sourcegraph/internal/k8s/resource/service"
+	"github.com/sourcegraph/sourcegraph/internal/k8s/resource/serviceaccount"
+	"github.com/sourcegraph/sourcegraph/internal/k8s/resource/statefulset"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"github.com/sourcegraph/sourcegraph/lib/pointers"
+)
+
+func (r *Reconciler) reconcileCodeIntel(ctx context.Context, sg *config.Sourcegraph, owner client.Object) error {
+	if err := r.reconcileCodeIntelStatefulSet(ctx, sg, owner); err != nil {
+		return err
+	}
+	if err := r.reconcileCodeIntelPersistentVolumeClaim(ctx, sg, owner); err != nil {
+		return err
+	}
+	if err := r.reconcileCodeIntelConfigMap(ctx, sg, owner); err != nil {
+		return err
+	}
+	if err := r.reconcileCodeIntelSecret(ctx, sg, owner); err != nil {
+		return err
+	}
+	if err := r.reconcileCodeIntelService(ctx, sg, owner); err != nil {
+		return err
+	}
+	if err := r.reconcileCodeIntelServiceAccount(ctx, sg, owner); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Reconciler) reconcileCodeIntelStatefulSet(ctx context.Context, sg *config.Sourcegraph, owner client.Object) error {
+	cfg := sg.Spec.CodeIntel
+	name := "codeintel-db"
+
+	ctrImage, err := config.GetDefaultImage(sg, name)
+	if err != nil {
+		return err
+	}
+
+	ctr := container.NewContainer(name, cfg, config.ContainerConfig{
+		Image: ctrImage,
+		Resources: &corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceRequestsCPU: resource.MustParse("4"),
```

Comment:
```suggestion
				corev1.ResourceCPU: resource.MustParse("4"),
```

And similar in a couple of other lines. This is what's causing `requests.cpu: ...` to appear in the compare-helm diff, instead of just `cpu: ...`.

The thing is... I'm surprised this creates a valid kubernetes resource. The fact that this resource appears in the golden fixtures shows that the resultant manifest was valid according the the k8s schema for it. Weird 🤔

Reply:
Nice catch, I missed that, but yeah.. that is weird.